### PR TITLE
Clean up json encoding for chemistry magic

### PR DESCRIPTION
**Pull request by anpaz-msft**

### CODE REVIEW

File: Chemistry/src/DataModel/LadderOperator/LadderOperator.cs [hunk 1]
```diff
@@ -10,6 +11,7 @@ namespace Microsoft.Quantum.Chemistry.LadderOperators
     /// <summary>
     /// Data strcture for raising and lowering operators.
     /// </summary>
+    [JsonConverter(typeof(LadderOperator.JsonConverter))]
     public struct LadderOperator : IEquatable<LadderOperator>
```
Comment (guanghaolow) [inline Chemistry/src/DataModel/LadderOperator/LadderOperator.cs]: Note that LadderOperators are now generic in the update #66. However, only specializations LadderOperator<Int> (for current Hamiltonians) and LadderOperator<SpinOrbital> (for wavefunctions) are currently use.

File: Chemistry/src/DataModel/LadderOperator/LadderSequence.cs [hunk 2]
```diff
@@ -4,13 +4,15 @@
...
 
 namespace Microsoft.Quantum.Chemistry.LadderOperators
 {
-    
+
     /// <summary>
     /// Class representing a sequence of raising and lowering operators.
     /// </summary>
+    [JsonConverter(typeof(LadderSequence.JsonConverter))]
     public class LadderSequence : IEquatable<LadderSequence>
```
Comment (guanghaolow) [inline Chemistry/src/DataModel/LadderOperator/LadderSequence.cs]: See [comment](https://github.com/Microsoft/QuantumLibraries/pull/67/files#r274252989)

File: Chemistry/src/DataModel/TermTypes.cs [hunk 3]
```diff
@@ -88,6 +97,7 @@ public enum RaisingLowering
     /// Boxed version of double that implements the <see cref="ITermValue"></see> interface
     /// that requires all values to have a method to compute its norm.
     /// </summary>
```
Comment (guanghaolow) [inline Chemistry/src/DataModel/TermTypes.cs]: SpinOrbital.Indexconvention is moved to this file in the update

**Comment by guanghaolow** [comment]
Looks good!

File: Chemistry/src/DataModel/LadderOperator/LadderOperator.cs (hunk 1 above)
Comment (anpaz-msft) [reply Chemistry/src/DataModel/LadderOperator/LadderOperator.cs]: That might be problematic for this scenario, as I believe you can't specify a generic `JsonConverter<>`.
What we will need to do is to keep the `JsonConverter` generic but remove it from the base `LadderOperator` and then each concrete `LadderOperator` will have to specify its own `JsonConverter` attribute.

**Pull request merged**